### ISSUE TITLE
AWS Toolkit Debugger Issue

### ISSUE THREAD

**Issue by Rach81** [aws/aws-toolkit-vscode#1710]
I have been successfully debugging my lambda functions locally for the past year and a half, using AWS Toolkit in Visual Studio Code.

Today I installed Jest to figure out how to unit test with this framework and now I cannot debug in any of my applications. When I try, it gives me this error.

Error: [InvalidResourceException('', 'Logical ids must be alphanumeric.')] ('', 'Logical ids must be alphanumeric.')

It seems like it wants me to set the configuration in my launch config with the resource name of the lambda function in the template as the target, as demonstrated below.

```
 "configurations": [
        {
            "type": "aws-sam",
            "request": "direct-invoke",
            "name": "testDebugger:HelloWorldFunction (nodejs12.x)",
            "invokeTarget": {
                "target": "template",
                "templatePath": "${workspaceFolder}/testDebugger/template.yaml",
                "logicalId": "HelloWorldFunction"
            },
            "lambda": {
                "payload": {},
                "environmentVariables": {},
                "runtime": "nodejs12.x"
            }
        },
```
My current configurations have the target set to "code".

```
"configurations": [
        {
            "type": "aws-sam",
            "request": "direct-invoke",
            "name": "<the name> (nodejs12.x)",
            "invokeTarget": {
                "target": "code",
                "projectRoot": "<the project root>",
                "lambdaHandler": "app.lambdaHandler"
            },
            "lambda": {
                "runtime": "nodejs12.x",
                "payload": {},
                "environmentVariables": {}
            }
        }
```

When I try to debug, it gives me these errors.

2021-05-03 09:33:12 [WARN]: "sam build" failed: <path>/app___vsctk___template.yaml
2021-05-03 09:33:12 [INFO]: Starting SAM application locally
2021-05-03 09:33:13 [INFO]: AWS.running.command
Error: [InvalidResourceException('my-lambda-function', 'Logical ids must be alphanumeric.')] ('my-lambda-function', 'Logical ids must be alphanumeric.')

Command stopped: "sam local invoke"
2021-05-03 09:33:15 [ERROR]: Timeout while waiting for command: "sam local invoke"
2021-05-03 09:33:15 [ERROR]: Failed to run SAM application locally: Timeout while waiting for command: "sam local invoke"
2021-05-03 09:33:15 [INFO]: Waiting for SAM application to start...
2021-05-03 09:33:45 [WARN]: Failed to use debugger port: 5858
2021-05-03 09:33:45 [INFO]: Attaching debugger to SAM application...

As I said, It seems to be trying to do a direct invoke on my template when my launch file is set to invoke the code. This has always worked up until now. Does anyone know what might be causing this or how I can resolve it?

**Comment by bryceitoc9** [comment]
Hi @Rach81 

This was a regression from the last build. Fix here: https://github.com/aws/aws-toolkit-vscode/pull/1687 , already staged for the next release.

If you'd like to get unblocked immediately, feel free to use the artifact here: https://d1ihu6zq92vp9p.cloudfront.net/0161b9fe-757e-4b1f-8810-718e6b284506/report.html ; this is from the latest commit on the `master` branch.

**Comment by Rach81** [comment]
> Hi @Rach81
> 
> This was a regression from the last build. Fix here: #1687 , already staged for the next release.
> 
> If you'd like to get unblocked immediately, feel free to use the artifact here: https://d1ihu6zq92vp9p.cloudfront.net/0161b9fe-757e-4b1f-8810-718e6b284506/report.html ; this is from the latest commit on the `master` branch.

I used the artifact for now and that resolved my issue. Thank you so much for the prompt reply.
